libvast/src/system/default_configuration.cpp

```diff
@@ -18,14 +18,32 @@
 
 #include <caf/config_value.hpp>
 
+#include <chrono>
+
 namespace vast::system {
 
 default_configuration::default_configuration() {
   // Tweak default logging options.
   using caf::atom;
+  using namespace std::chrono_literals;
   set("logger.component-blacklist",
       caf::make_config_value_list(atom("caf"), atom("caf_flow"),
                                   atom("caf_stream")));
+  // TODO: Move part of this text into the commit message
+  // The `max-batch-delay` is the maximum amount of time that caf waits until
+  // it attempts to send an underfull batch along the stream.
+  // However, there is a possible deadlock: [...]
+  //  1) Increase `stream.max-batch-delay`. As long as the total cost
+  //  2) Set a finite, small `scheduler.max-throughput`. This should empirically
+  //     be enough, but it relies on the CAF scheduler to actually schedule the
+  //     downstream actor, which empirically can take quite a while at times.
+  //     Also, if this is too small, there is a chance that a very busy stream
+  //     can drown out an actor completely so that no other work can take place.
+  //  3) Increase `scheduler.max-threads`. When there are more threads than
+  //     deadlocked actors, eventually the downstream actors will be able to
+  //     make progress and the deadlock can be resolved.
```

Comment:
The formulation in the comment is confusing, do you need to adjust the settings as described in 1,2,3 for the deadlock to surface or to mitigate it?